proto/tendermint/types/ipaddress.go

```diff
@@ -0,0 +1,166 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+
+	"inet.af/netaddr"
+)
+
+type IPAddress struct {
+	data netaddr.IP
+}
+
+// Returns underlying netaddr.IP
+func (ip IPAddress) ToNetaddrIP() netaddr.IP {
+	return ip.data
+}
+
+// Parse parses provided IP and sets its value to itself. Supports IPv4 and IPv6.
+func (ip *IPAddress) Parse(address string) error {
+	addr, err := netaddr.ParseIP(address)
+	if err != nil {
+		return err
+	}
+	// Convert ipv4 to ipv6 form without unmapping, as that's what we store
+	if addr.Is4() {
+		v6 := addr.As16()
+		addr = netaddr.IPv6Raw(v6)
+	}
+	ip.data = addr
+	return nil
+}
+
+// MustParse parses provided address and returns itself.
+// It will panic on error.
+// Useful for chaining in tests.
```

Comment:
what do you think if to define this function in a test file, in a case we are going to use this only there?

Reply:
OK, it's bad practice to use it, will move to _test.